client/src/context/auth-context.js

```diff
@@ -1,89 +1,198 @@
+          Authorization: 'Bearer ' + String(authTokens.access_token),
+        },
+        body: JSON.stringify({
+          email: e.target.email.value,
+          password: e.target.password.value,
+        }),
+      }
+    );
+    let data = await response.json();
+
+    if (response.status === 200) {
+      // this is the check code !! 🤞
+      setAuthTokens(data);
+      setUser(jwt_decode(data.access_token));
+      localStorage.setItem('authTokens', JSON.stringify(data)); // here is the code that will be used to storedToken
+      history.push('/profile'); // this will send the user to the profile page the moment he logs in
+    } else {
+      alert('Something went wrong!');
+    }
   };
-};
-
-export const AuthContextProvider = (props) => {
-  const tokenData = retrieveStoredToken();
-
-  let initialToken;
-  if (tokenData) {
-    initialToken = tokenData.token;
-  }
 
-  const [token, setToken] = useState(initialToken);
-
-  const userIsLoggedIn = !!token;
-
-  const logoutHandler = useCallback(() => {
-    setToken(null);
-    localStorage.removeItem('token');
-    localStorage.removeItem('expirationTime');
+  let logoutUser = () => {
+    // this is for the time we are logged out
+    setAuthTokens(null);
+    setUser(null);
+    localStorage.removeItem('authTokens');
+    history.push('/login');
+  };
 
-    if (logoutTimer) {
-      clearTimeout(logoutTimer);
+  const updateToken = async () => {
+    let response = await fetch(
+      'http://127.0.0.1:8000/api/auth/generatetokens',
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: 'Bearer ' + String(authTokens.refresh_token),
+        },
+        body: JSON.stringify({ refresh_token: authTokens?.refresh_token }),
+      }
+    );
+
+    let data = await response.json();
+
+    if (response.status === 200) {
+      setAuthTokens(data);
+      setUser(jwt_decode(data.access_token));
+      localStorage.setItem('authTokens', JSON.stringify(data));
+    } else {
+      logoutUser();
     }
-  }, []);
 
-  const loginHandler = (token, expirationTime) => {
-    setToken(token);
-    localStorage.setItem('token', token);
-    localStorage.setItem('expirationTime', expirationTime);
-
-    const remainingTime = calculateRemainingTime(expirationTime);
+    if (loading) {
+      setLoading(false);
+    }
+  };
 
-    logoutTimer = setTimeout(logoutHandler, remainingTime);
+  let contextData = {
+    // here we can access the data from the server
+    user: user,
+    authTokens: authTokens,
+    loginUser: loginUser,
+    logoutUser: logoutUser,
   };
 
   useEffect(() => {
-    if (tokenData) {
-      logoutTimer = setTimeout(logoutHandler, tokenData.duration);
+    if (loading) {
+      updateToken();
     }
-  }, [tokenData, logoutHandler]);
 
-  const contextValue = {
-    token: token,
-    isLoggedIn: userIsLoggedIn,
-    login: loginHandler,
-    logout: logoutHandler,
-  };
+    let onehrs = 1000 * 60 * 60;
```

Comment:
Please use camCasing for defining variables.

Reply:
DONE